code/modules/security_levels/keycard_authentication.dm

```diff
@@ -101,6 +101,10 @@ GLOBAL_DATUM_INIT(keycard_events, /datum/events, new)
 	event_source = null
 
 /obj/machinery/keycard_auth/proc/trigger_event(confirmer)
+	if(confirmer == triggerer)
+		message_admins("[ADMIN_LOOKUPFLW(triggerer)] attempted to both trigger and confirm event [event].")
```

Comment:
does this really need to message admins?

Reply:
well you know, fucking cheaters innit

Reply:
well you know, fuckin cheaters innit